filebeat/input/filestream/input.go

```diff
@@ -359,16 +368,24 @@ func (inp *filestream) eventFromMessage(m reader.Message, path string) beat.Even
 		},
 	}
 	fields.DeepUpdate(m.Fields)
+	m.Fields = fields
+
+	for _, proc := range inp.msgPostProc {
+		proc.PostProcess(&m)
+	}
```

Comment:
Why do we need to build and apply the post processors separately? Couldn't we just ensure that the parser correctly modifies the `message`? Are there fields that must be 'set' before the post processing is run?

Reply:
Yes, the fields `log.offset`, `log.file.path` are set before the input merges the JSON fields with the existing fields.